python/lsst/pipe/supertask/cmdLineParser.py

```diff
@@ -261,16 +251,19 @@ def makeParser(fromfile_prefix_chars='@', parser_class=ArgumentParser, **kwargs)
     group = parser.add_argument_group("Data repository and selection options")
     group.add_argument("-b", "--butler-config", dest="butler_config", default=None, metavar="PATH",
                        help="Location of the gen3 butler/registry config file.")
-    group.add_argument("-i", "--input", dest="input",
-                       metavar="COLLECTION", default=None,
-                       help=("Name of the data butler collection used for "
-                             "input, overrides collection specified in Butler "
-                             "configuration file"))
-    group.add_argument("-o", "--output", dest="output",
-                       metavar="COLLECTION", default=None,
-                       help=("Name of the data butler collection used for "
-                             "output, overrides collection specified in Butler "
-                             "configuration file"))
+    group.add_argument("-i", "--input", dest="input", type=_inputCollectionType,
+                       metavar="COLL,DSTYPE:COLL", default={},
+                       help=("Comma-separated names of the data butler collection. "
+                             "If collection includes dataset type name separated by colon "
+                             "then collection is only used for that specific dataset type. "
+                             "Pre-flight uses these collections to search for input datasets. "
+                             "Task execution stage only uses first global collection name "
+                             "to override collection specified in Butler configuration file."))
+    group.add_argument("-o", "--output", dest="output", type=_outputCollectionType,
+                       metavar="COLL,DSTYPE:COLL", default={},
+                       help=("Comma-separated names of the data butler collection. "
+                             "See description of --input option.  This option only allows "
+                             "single collection (per-dataset type or global)."))
```

Comment:
I forgot to mention this originally, but the `Butler` itself will not actually permit multiple output collections in execution (even if there is only one output collection per DatasetType).  So while it's okay to leave support for that in the daf_butler Registry code in case it is useful in the future, the argument parser here should probably not present an interface that suggests that we can support multiple output collections right now.

Reply:
Butler indeed does not do that but pre-flight now allows collection-per-DatasetType  (multiple collections for inputs but only single collection for output DatasetType). If this is not useful idea then we can just say that there is one output collection for all DatasetTypes and change this option and `DatasetOriginInfo` interface.